filebeat/harvester/log.go

```diff
@@ -104,24 +114,50 @@ func (h *Harvester) Harvest() {
 		// Reset Backoff
 		h.backoff = h.Config.BackoffDuration
 
-		// Sends text to spooler
-		event := &input.FileEvent{
-			ReadTime:     lastReadTime,
-			Source:       &h.Path,
-			InputType:    h.Config.InputType,
-			DocumentType: h.Config.DocumentType,
-			Offset:       h.Offset,
-			Bytes:        bytesRead,
-			Text:         &text,
-			Fields:       &h.Config.Fields,
-			Fileinfo:     &info,
+		if h.shouldExportLine(text) {
+
+			// Sends text to spooler
+			event := &input.FileEvent{
+				ReadTime:     lastReadTime,
+				Source:       &h.Path,
+				InputType:    h.Config.InputType,
+				DocumentType: h.Config.DocumentType,
+				Offset:       h.Offset,
+				Bytes:        bytesRead,
+				Text:         &text,
+				Fields:       &h.Config.Fields,
+				Fileinfo:     &info,
+			}
+
+			event.SetFieldsUnderRoot(h.Config.FieldsUnderRoot)
+			h.SpoolerChan <- event // ship the new event downstream
 		}
 
+		// Set Offset
 		h.Offset += int64(bytesRead) // Update offset if complete line has been processed
+	}
+}
 
-		event.SetFieldsUnderRoot(h.Config.FieldsUnderRoot)
-		h.SpoolerChan <- event // ship the new event downstream
+// shouldExportLine decides if the line is exported or not based on
+// the include_lines and exclude_lines options.
+func (h *Harvester) shouldExportLine(line string) bool {
```

Comment:
Is it on purpose that both include_lines and exclude_lines can both be set? We should probably mention the order it is checked in the docs.

In case a line is matched by include_line and exclude_line, the line is excluded.